src/Hl7.Fhir.Base/Utility/CodedWithLocationException.cs

```diff
@@ -0,0 +1,97 @@
+﻿/* 
+ * Copyright (c) 2021, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://raw.githubusercontent.com/FirelyTeam/firely-net-sdk/master/LICENSE
+ */
+
+using Hl7.Fhir.Model;
+using System;
+
+#nullable enable
+
+namespace Hl7.Fhir.Utility
+{
+    /// <summary>
+    /// This is a class of Exceptions that is raised by the SDK and is coded with a unique code
+    /// to enable callers to identify this exception and react appropriately on the code.
+    /// </summary>
+    /// <remarks>Most modules of the SDK using this Exception will create specific subclasses
+    /// for this subclass, providing a list of codes used by that module.</remarks>
+    public class CodedWithLocationException : CodedException
```

Comment:
Should we call this: `OperationOutcomeCodedException`, because that's basically what it is: a CodedException with enough information (not just the location!!) to produce an OperationOutcome from?

Reply:
Then the ToIssue member makes sense, since that presents one of the principal purposes of the OperationOutcomeCodedException.

Reply:
How about a rename to `ExtendedCodedException`?